recipes/default.rb

```diff
@@ -24,9 +24,20 @@
 # ::Chef::Recipe.send(:include, Chef::Mixin::PowershellOut)
 ::Chef::Resource::RubyBlock.send(:include, Chef::Mixin::PowershellOut)
 
-powershell_script 'install chocolatey' do
-  code "iex ((new-object net.webclient).DownloadString('#{node['chocolatey']['Uri']}'))"
-  convert_boolean_return true
+if File.exist?('C:\windows\sysnative\cmd.exe')
+  arch = :x86_64
+  cmd = 'C:\windows\sysnative\cmd.exe'
+else
+  arch = nil
+  cmd = 'cmd.exe'
+end
+
+batch 'install chocolatey' do
+  architecture arch
+  interpreter cmd
+  code <<-EOH
+    powershell -noprofile -inputformat none -noninteractive -executionpolicy bypass -command "iex ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1'))"
```

Comment:
This seems to have removed `#{node['chocolatey']['Uri']}`, which I might imagine is still important?